### DISCUSSION TITLE
SSL via PEM Support

### DISCUSSION THREAD

**Discussion by dirkkuyken** [provectus/kafka-ui#2976] [Q&A] [answered]
Hi all,

I have a Kafka cluster making use of PEM + KEY file for its SSL setup:
`      
      KAFKA_SECURITY_PROTOCOL: SSL
      KAFKA_SSL_KEYSTORE_TYPE: PEM
      KAFKA_SSL_KEYSTORE_CERTIFICATE_CHAIN: 
      KAFKA_SSL_KEYSTORE_KEY:`

The Kafka server successfully started, now I want to connect the Kafka UI to it making use of the SSL setup, but the documentation is only mentioning .jks.
Is there any support for PEM in the UI? 

Over PLAINTEXT it obviously works, but I need to have it secured as the UI will be running in a different location (UI in AWS towards Kafka on prem)

**Comment by Haarolean** [comment, marked as answer]
Hey, yeah, that's possible.
```
KAFKA_CLUSTERS_0_PROPERTIES_SSL_KEYSTORE_TYPE: 'PEM'
```
and other properties you'd need have to be formatted the same way.
Some of thme (ssl related) you can find [here](https://github.com/a0x8o/kafka/blob/master/clients/src/main/java/org/apache/kafka/common/config/SslConfigs.java)
